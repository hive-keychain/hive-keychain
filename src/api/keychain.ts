import axios from 'axios';

const KeychainApi = axios.create({
  baseURL:
    process.env.KEYCHAIN_API_DEV === 'true'
      ? 'http://localhost:3000/'
      : 'https://api.hive-keychain.com/',
});

export default KeychainApi;
