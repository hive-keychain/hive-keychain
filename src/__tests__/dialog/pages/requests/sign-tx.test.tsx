import { KeychainKeyTypes } from '@interfaces/keychain.interface';
import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render } from '@testing-library/react';
import React from 'react';
import SignTx from 'src/dialog/pages/requests/sign-tx';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import requestSignTx from 'src/__tests__/utils-for-testing/data/props/dialog/request-signTx';
import objects from 'src/__tests__/utils-for-testing/helpers/objects';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
import { PropsRequestsSignTx } from 'src/__tests__/utils-for-testing/types/props-types';
describe('sign-tx tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestSignTx;
  methods.config();
  it('Must show SignTx dialog(canwhitelist)', async () => {
    const clonedProps = objects.clone(props) as PropsRequestsSignTx;
    clonedProps.data.method = KeychainKeyTypes.posting;
    const { asFragment } = render(<SignTx {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot('Request SignTx canwhitelist');
  });
  it('Must show SignTx dialog(no whitelisting)', async () => {
    const clonedProps = objects.clone(props) as PropsRequestsSignTx;
    clonedProps.data.method = KeychainKeyTypes.active;
    const { asFragment } = render(<SignTx {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot('Request SignTx no whitelisting)');
  });
  it('Must call window.close when clicking on cancel', async () => {
    const clonedProps = objects.clone(props) as PropsRequestsSignTx;
    clonedProps.data.method = KeychainKeyTypes.active;
    render(<SignTx {...clonedProps} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    const clonedProps = objects.clone(props) as PropsRequestsSignTx;
    clonedProps.data.method = KeychainKeyTypes.active;
    render(<SignTx {...{ ...clonedProps, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    const clonedProps = objects.clone(props) as PropsRequestsSignTx;
    clonedProps.data.method = KeychainKeyTypes.active;
    render(<SignTx {...clonedProps} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.sendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: clonedProps.data,
        tab: clonedProps.tab,
        domain: clonedProps.domain,
        keep: false,
      },
    });
  });
});
