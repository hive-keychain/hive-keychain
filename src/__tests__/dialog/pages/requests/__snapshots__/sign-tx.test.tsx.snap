// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign-tx tests:
 Must show SignTx dialog(canwhitelist): Request SignTx canwhitelist 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Sign Transaction
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Key
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        Posting
      </div>
      <h3
        aria-label="clickeable-header"
      >
        Transaction (click to show)
      </h3>
      <div
        aria-label="collapsible-div"
        class="hide"
      >
        <div
          class="operation_item_content"
        >
          <pre
            role="contentinfo"
          >
            {
  "ref_block_num": 1000,
  "ref_block_prefix": 1,
  "expiration": "12/12/2023",
  "operations": [],
  "extensions": []
}
          </pre>
        </div>
      </div>
    </div>
    <div
      class="operation_footer "
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      >
        <div
          class="checkbox-container align-center"
        >
          <div
            class="checkbox-and-label"
          >
            <div
              class="pretty p-default"
            >
              <input
                aria-checked="false"
                id="checkbox-set-keep-inner-input"
                type="checkbox"
                value=""
              />
              <div
                class="state"
              >
                <label
                  for="checkbox-set-keep-inner-input"
                />
              </div>
            </div>
            <div
              aria-label="checkbox-set-keep"
              class="label"
            >
              Do not prompt again to send signTx transactions from the @keychain.tests account on domain
            </div>
          </div>
        </div>
      </div>
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`sign-tx tests:
 Must show SignTx dialog(no whitelisting): Request SignTx no whitelisting) 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Sign Transaction
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Key
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        Active
      </div>
      <h3
        aria-label="clickeable-header"
      >
        Transaction (click to show)
      </h3>
      <div
        aria-label="collapsible-div"
        class="hide"
      >
        <div
          class="operation_item_content"
        >
          <pre
            role="contentinfo"
          >
            {
  "ref_block_num": 1000,
  "ref_block_prefix": 1,
  "expiration": "12/12/2023",
  "operations": [],
  "extensions": []
}
          </pre>
        </div>
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
